compiler/bindings/c/iree/compiler/embedding_api.h

```diff
@@ -117,10 +117,6 @@ IREE_EMBED_EXPORTED void ireeCompilerSetupGlobalCL(int argc, const char **argv,
 // This must be called prior to library unloading.
 IREE_EMBED_EXPORTED void ireeCompilerGlobalShutdown();
 
-// Invokes a callback with each registered HAL target backend.
```

Comment:
This is a versioned API. At a minimum, we need to keep this entry-point as a no-op and add the new one with a new version. See how it is done for ireeCompilerEnumeratePlugins.

However, this API is really just a kludge. I think I would prefer that it just internally create an invocation/session/activate plugins and then enumerate. We can update the comment to indicate that it does so and is really only suitable for global CLI-like tools.

Reply:
Gotcha. I reverted the breaking changes and moved the session creation and plugin activation into the existing API implementation. PTAL.